packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -377,110 +637,146 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     options?: InputGenerateTransactionOptions
   ): Promise<AccountAuthenticator> {
     try {
-      this.doesWalletExist();
-      // if input is AnyRawTransaction, i.e V2
-      if ("rawTransaction" in transactionOrPayload) {
-        if (this._wallet?.version !== "v2") {
-          throw new WalletNotSupportedMethod(
-            `Sign Transaction V2 is not supported by ${this.wallet?.name}`
-          ).message;
+      this.ensureWalletExists(this._wallet);
+      this.ensureAccountExists(this._account);
+
+      // Make sure wallet supports signTransaction
+      if (this._wallet.signTransaction) {
+        // If current connected wallet is AIP-62 standard compatible
+        // we want to make sure the transaction input is what the
+        // standard expects, i,e new sdk v2 input
+        if (this._wallet.isAIP62Standard) {
+          // if rawTransaction prop it means transaction input data is
+          // compatible with new sdk v2 input
+          if ("rawTransaction" in transactionOrPayload) {
+            return await this.walletStandardCore.signTransaction(
+              transactionOrPayload,
+              this._wallet,
+              asFeePayer
+            );
+          } else {
+            // else it means dapp passes legacy sdk v1 input data
+            // and the current connected wallet does not support it
+            throw new WalletNotSupportedMethod(
+              `Wrong transaction input data was porivded, ${this.wallet?.name} expects AnyRawTransaction input type. 
+              Please upgrade to the new Aptos TS SDK`
```

Comment:
A link might be more helpful as we push for more aggressive deprecation.

```suggestion
              Please upgrade to the latest Aptos TS SDK https://github.com/aptos-labs/aptos-ts-sdk`
```